src/Types/Input/NonNullableInputType.hack

```diff
@@ -0,0 +1,39 @@
+namespace Slack\GraphQL\Types;
+
+use namespace Slack\GraphQL;
+use namespace Graphpinator\Parser\Value;
+
+final class NonNullableInputType<+TInner as nonnull>
+    extends InputType<TInner>
+    implements GraphQL\Introspection\IntrospectableGeneric {
+
+    public function __construct(private InputType<TInner> $inner_type) {}
+
+    <<__Override>>
+    public function getName(): ?string {
+        return null;
+    }
+
+    <<__Override>>
+    public function coerceValue(mixed $value): TInner {
+        return $this->inner_type->coerceValue($value);
+    }
+
+    <<__Override>>
+    final public function coerceNonVariableNode(Value\Value $node, dict<string, mixed> $variable_values): TInner {
+        return $this->inner_type->coerceNode($node, $variable_values);
+    }
+
+    <<__Override>>
+    public function assertValidVariableValue(mixed $value): TInner {
+        return $this->inner_type->assertValidVariableValue($value);
+    }
+
+    public function getTypeKind(): GraphQL\Introspection\__TypeKind {
```

Comment:
I needed to add these NonNullable types so we could output the NOT_NULL type kind.

Reply:
Hmm we only need these wrappers for introspection, we should be able to avoid using them elsewhere (e.g. during execution). That could improve performance (less wrappers) and also simplify the code (e.g. we won't need these `coerce...` methods there because that's not something that introspection needs).

This would mean that there are 2 representations of each type, 1 for runtime (what we have now) and 1 for introspection, so e.g. FieldDefinition will have `getType()` that returns the normal type and `getIntrospectionType()` which returns the type wrapped in a NonNullable wrapper. So we pay for the simplified code here by making a few other places more complicated, but overall I think it should still end up being simpler...?